app/models/concerns/interaction_tracking.rb

```diff
@@ -0,0 +1,18 @@
+module InteractionTracking
+  extend ActiveSupport::Concern
+
+  # When a client contacts us, flag client for attention and update last incoming response.
+  def record_incoming_touch
```

Comment:
Non-blocking: I'd prefer some renames:

- `record_incoming_touch` -> `record_incoming` or `record_incoming_message` or `record_incoming_interaction`.
- `record_outgoing_touch` -> `record_outgoing` or `record_outgoing_message`.
- `record_internal_touch` -> `record_internal` or `record_internal_message` or `record_internal_note`.

I find the word `touch` in the method name confusing.